src/tests/ftest/pool/evict_test.py

```diff
@@ -38,110 +39,257 @@ class EvictTest(TestWithServers):
 
     def test_evict(self):
         """
-        Test evicting a client to a pool.
+        Test evicting a client from a pool.
 
         :avocado: tags=pool,poolevict,quick
         """
-        size = self.params.get("size", '/run/tests/sizes/size1gb/*')
-        setid = self.params.get("setname", '/run/tests/setnames/validsetname/*')
-        connectperm = self.params.get("perms",
-                                      '/run/tests/connectperms/permro/*')
+        pool = []
+        container = []
+        cont_uuid = []
+        non_pool_servers = []
+        # Target list is configured so that the pools are across all servers
+        # except the pool under test is created on half of the servers
+        pool_tgt = [num for num in range(len(self.hostlist_servers))]
+        pool_tgt_ut = [num for num in range(int(len(self.hostlist_servers)/2))]
+        tlist = [pool_tgt, pool_tgt, pool_tgt_ut]
+        pool_servers = [self.hostlist_servers[:len(tgt)] for tgt in tlist]
+        # Create TestPool
+        for count, _ in enumerate(tlist):
+            self.log.info("Create a pool")
+            pool.append(TestPool(self.context, self.log))
+            pool[count].get_params(self)
+            pool[count].name.value = self.server_group
+            pool[count].target_list.value = tlist[count]
+            self.log.info(
+                "Pool # %s target list is %s",
+                count+1, pool[count].target_list)
+            pool[count].create()
+            self.log.info("Pool # %s UUID is %s and handle is %s",
+                          count+1, pool[count].uuid,
+                          pool[count].pool.handle.value)
+
+            # Check that the pools exist
+            self.assertTrue(
+                pool[count].check_files(pool_servers[count]),
+                "Pool # {} data not detected on servers before evict".format(
+                    count+1))
+            # Check that pool does not exists on targets not in pool
+            for host in self.hostlist_servers:
+                if host not in pool_servers[count]:
+                    non_pool_servers.append(host)
+            if len(non_pool_servers) > 0:
+                self.assertFalse(
+                    pool[count].check_files(non_pool_servers),
+                    "Pool # {} data detected on non pool servers {} ".format(
+                        count+1, non_pool_servers))
+
+            # Connect to each pool
+            self.assertTrue(pool[count].connect(1 << 1),
```

Comment:
The shift is done inside of TestPool.connect()
```suggestion
            self.assertTrue(pool[count].connect(1),
```